File: Algorithm/Algorithmic Problem Solving Strategies/C++/6.1.cpp
#include <iostream>

using namespace std;

class RecursiveSum
{
public:
	int sum(int n);
	int recursiveSum(int n);
};


int RecursiveSum::sum(int n)
{
	int ret = 0;
	for (int i = 1; i <= n; ++i)
	{
		ret += i;
	}

	return ret;
}


int RecursiveSum::recursiveSum(int n)
{
	if (n == 1) 
	{
		return 1;
	}

	return n + recursiveSum(n - 1);
}



int main()
{
	int n;
	cin >> n;

	RecursiveSum recursive_sum;
	cout << recursive_sum.sum(n) << endl;
	cout << recursive_sum.recursiveSum(n) << endl;
}

File: Algorithm/Algorithmic Problem Solving Strategies/Go/6.1.go
// 6.1
package main

import (
	"fmt"
)


type RecursiveSum struct {
}

func (r RecursiveSum) sum(n int) int {
	ret := 0

	for i := 0; i <= n; i++ {
		ret += i
	}

	return ret
}

func (r RecursiveSum) recursiveSum(n int) int {

	if n == 1 {
		return 1
	}

	return n + r.recursiveSum(n-1)
}



func main() {

	var n int
	fmt.Scanln(&n)

	recursive_sum := RecursiveSum{}
	fmt.Println(recursive_sum.sum(n))
	fmt.Println(recursive_sum.recursiveSum(n))
}
